### ISSUE TITLE
Missing item names in several rows

### ISSUE THREAD

**Issue by master-nevi** [philipmulcahy/azad#291]
Hello, I use this extension for 2 accounts (mine and my partners) for almost a year now. I noticed recently that my account does not succeed in scrapping and showing all the order items anymore yet other order data like total and orderID show just fine. Here's a screenshot:

`1m`:
![missing-items-screenshot-1m](https://github.com/user-attachments/assets/61ccbad9-f661-48cb-8e7c-1e6947519331)

`3m`:
![missing-items-screenshot-3m](https://github.com/user-attachments/assets/32641060-5ecf-49c4-9a0c-b132dcd3f16d)

I always click `1m` or the last month's worth of orders. When I try `3m` I get every order I was able to scrape in the past. It's almost like only new orders as of this month have started to not show the item name.

I have emailed the json files.

**Comment by philipmulcahy** [comment]
1) do you get the popup that there is a login problem?2) have you tried clearing azad’s cache?(asking because the items info comes from the “detail” page for the order, unlike the basic fields, and sometimes Amazon can log you out on the detail pages without the message getting to the basic part of the site)3) if no to 1 and yes to 2 and it still fails then the next thing is to collect a console log of such a failed scrape (immediately after clearing azad cache) and then getting an order debug dump json file for an impacted order (see the readme on how to do both)Yours PhilipSent from my phoneOn 20 Sep 2024, at 17:58, David Robles ***@***.***> wrote:﻿
Hello, I use this extension for 2 accounts (mine and my partners) for almost a year now. I noticed recently that my account does not succeed in scrapping and showing all the order items anymore yet other order data like total and orderID show just fine. Here's a screenshot:
missing-items-screenshot.jpg (view on web)

**Comment by master-nevi** [comment]
Omg, clearing the cache seems to have fixed the problem. Thank you. I
wonder if this should be automatic somehow, if the button to clear cache
should be available on the table view.

**Comment by philipmulcahy** [comment]
Good!
I’ve considered exactly that approach (making cache clearance UI more prominent), but it has a big downside: the reason for the cache existing at all is that some large accounts can’t scrape a year without getting logged out by Amazon (scraper detection). The cache enables them to log back in and make incremental progress, rather than starting from zero again. If people are too trigger happy with clearing the cache, they’ll moan about not being able to get their year (believe me, they already do so).
In general, most user issues (including yours) are dealt with by reading the README, and when I am particularly overloaded I close tickets preemptively if folks have not signalled that they’ve read it <https://github.com/philipmulcahy/azad?tab=readme-ov-file#common-problems-with-some-fixes>

**Comment by master-nevi** [comment]
@philipmulcahy what if for these cases, entries don't get cached if table rows are missing data. Is something like that possible?

**Comment by master-nevi** [comment]
Hmm, I actually don't think this has to do with the cache. It's still happening frequently despite having cleared the cache. I can also see in the extension page that there's a cache-hit of `0`; which I would think means it's not getting much out of the cache:
![Screenshot 2024-09-22 at 4 20 49 PM](https://github.com/user-attachments/assets/2749bf04-b722-4479-a57a-2daad79de82d)

**Comment by philipmulcahy** [comment]
Thanks for following up! Let’s reopen (if you are willing to help me investigate).I reckon that the same files would be the ideal starting point: a console log and a json dump file from an order that is missing its items.Sent from my phoneOn 23 Sep 2024, at 00:22, David Robles ***@***.***> wrote:﻿
Hmm, I actually don't think this has to do with the cache. It's still happening frequently despite having cleared the cache. I can also see in the extension page that there's a cache-hit of 0; which I would think means it's not getting much out of the cache:
Screenshot.2024-09-22.at.4.20.49.PM.png (view on web)

**Comment by justinhaaheim** [comment]
+1 for me. 

I just did a 3mo scrape, and 40/99 orders have nothing in the Items column. I cleared the cache, repeated the scrape and got the same result -- the same item fields are missing.

**Comment by master-nevi** [comment]
> Thanks for following up! Let’s reopen (if you are willing to help me investigate).I reckon that the same files would be the ideal starting point: a console log and a json dump file from an order that is missing its items.Sent from my phoneOn 23 Sep 2024, at 00:22, David Robles ***@***.***> wrote:﻿ Hmm, I actually don't think this has to do with the cache. It's still happening frequently despite having cleared the cache. I can also see in the extension page that there's a cache-hit of 0; which I would think means it's not getting much out of the cache: Screenshot.2024-09-22.at.4.20.49.PM.png (view on web) —Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>

Yep, I emailed the address in the README on Fri, Sep 20, 10:11 AM with all that stuff.

**Comment by philipmulcahy** [comment]
@master-nevi I found your email - thank you!
 It was in the spam folder and was not findable by a normal search (title:291) - I assume gmail doesn't like it because there's js embedded in the amazon pages that in turn are embedded in the json files.
I have what I need to start digging.

**Comment by kms54321** [comment]
Posted this under another issue, but thought I would post here as well:

I am having this problem as well. The extension worked perfectly for me two weeks ago, but when I tried it today - attempting to download ITEMS from the last month, 3 months, and year - in each case Amazon popped open a "sorry we couldn't find that page" tab. The scraped list then had gaps, with missing items spread throughout the table.

If I run the extension for ORDERS, I get a list showing all of my orders, but many of them show nothing in the ITEMS column, and several show only "ending in : ;" in the PAYMENTS column.

I've run the extension ensuring that only a single Amazon tab is open. I've also run it after being certain to have both cleared the extension's cache and purged my browsers cache and local storage for amazon.com. and have also run it successively without clearing either the extension cache or my browser cache, and I end up with the same results in all cases.

This extension has been very helpful to me, I thank you for creating it, and I have donated. I do not know how to get a "console log," but if I can figure that out then I will either post it or send it to you as appropriate.

Thanks!

**Comment by chlaird** [comment]
Same issue with a clean install today. Most rows are missing the items. I've tried clearing cache, no change. If I run the 3m report, 4 out of 50 have item names.

**Comment by philipmulcahy** [comment]
See progress (and engage) on #293 - it seems this issue, #292 , #293 and #294 are all caused by the same site update Amazon appears to be rolling out.
There's a test build I've shared there (that is an improvement, but has known defects). Expect either a second test build or a proper release next. I suspect I'm going to be able to fix pretty much all the problems except for transaction column in the shipping view, for which I've created #295.
